internal/clients/image_builder/image_client.go

```diff
@@ -0,0 +1,73 @@
+package image_builder
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/headers"
+	"github.com/RHEnVision/provisioning-backend/internal/parsing"
+)
+
+type ImageBuilderClient struct {
+	client *ClientWithResponses
+}
+
+func init() {
+	clients.GetImageBuilderClient = newImageBuilderClient
+}
+
+func newImageBuilderClient(ctx context.Context) (clients.ImageBuilder, error) {
+	c, err := NewClientWithResponses(config.ImageBuilder.URL)
+	if err != nil {
+		return nil, err
+	}
+	return &ImageBuilderClient{client: c}, nil
+}
+
+func (c *ImageBuilderClient) GetAWSAmi(ctx context.Context, composeID string) (string, error) {
+	ctxval.Logger(ctx).Info().Msgf("Getting AMI of image %v", composeID)
+	imageStatus, err := fetchImageStatus(ctx, c, composeID)
+	if err != nil {
+		return "", err
+	}
+	if imageStatus.Type != UploadTypesAws {
+		ctxval.Logger(ctx).Warn().Err(err).Msg("Image is not AWS type")
+		return "", BadImageTypeErr
+	}
+	awsStatus := imageStatus.Options.(AWSUploadStatus)
+	return awsStatus.Ami, nil
+}
+
+func fetchImageStatus(ctx context.Context, c *ImageBuilderClient, composeID string) (*UploadStatus, error) {
```

Comment:
```suggestion
func (c *ImageBuilderClient) fetchImageStatus(ctx context.Context, composeID string) (*UploadStatus, error) {
```

Reply:
It was a method before the` clients` refactoring, after I moved the `ImageBuilderClient` interface to the `clients` package, I had to define a `UploadStatus` struct in the clients model (as we do with `Source`) 
The `fetchImageStatus` is used internally and only by the client itself  (like a helper), therefore I removed it from the interface, and transformed it into a function implementation.

1. We can define `UploadStatus` struct in our client's models, but it's a bit complex, because it's dynamically changed according to the image type
2. We can create our own structure for the image status, though it might fail in a future regression
3. other suggestions? 🤔